### PR TITLE
Migrate Registration PIN screen to Mobius: Strangulate the controller

### PR DESCRIPTION
**Pull request by vinaysshenoy**
https://www.pivotaltracker.com/story/show/173283121

### CODE REVIEW

**Comment by vinaysshenoy** [comment]
Moving this back to `WIP` because I realised there's something I missed. Will fix.

**Comment by vinaysshenoy** [comment]
> Moving this back to `WIP` because I realised there's something I missed. Will fix.

Never mind, this was an issue in another screen, not in this one.

File: app/src/test/java/org/simple/clinic/registration/pin/RegistrationPinScreenControllerTest.kt [hunk 1]
```diff
@@ -126,6 +126,7 @@ class RegistrationPinScreenControllerTest {
   }
 
   private fun setupController(
+      requiredPinLength: Int = 4,
```
Comment (Honey14) [inline app/src/test/java/org/simple/clinic/registration/pin/RegistrationPinScreenControllerTest.kt:113]: Can we use the global constant `SECURITY_PIN_LENGTH` here?

File: app/src/test/java/org/simple/clinic/registration/pin/RegistrationPinScreenControllerTest.kt [hunk 2]
```diff
@@ -139,6 +140,7 @@ class RegistrationPinScreenControllerTest {
 
     val effectHandler = RegistrationPinEffectHandler(
         userSession = userSession,
+        schedulers = TrampolineSchedulersProvider(),
```
[the remark this answers is not in the document]
Comment (vinaysshenoy) [reply app/src/test/java/org/simple/clinic/registration/pin/RegistrationPinScreenControllerTest.kt:128]: Done.

File: app/src/test/java/org/simple/clinic/registration/pin/RegistrationPinScreenControllerTest.kt (hunk 1 above)
Comment (vinaysshenoy) [reply app/src/test/java/org/simple/clinic/registration/pin/RegistrationPinScreenControllerTest.kt:113]: Good catch, done!

**Pull request merged**